services/proxy/src/Proxy/API/Public.hs

```diff
@@ -65,14 +66,26 @@ import Wire.API.Routes.Public.Proxy
 type PublicAPI =
   ProxyAPIRoute "giphy-path" ("giphy" :> "v1" :> "gifs" :> RawM)
     :<|> ProxyAPIRoute "youtube-path" ("youtube" :> "v3" :> RawM)
-    -- :<|> ProxyAPIRoute "gmaps-static" ("googlemaps" :> "api" :> "staticmap" :> RawM)
+    :<|> ProxyAPIRoute
+           "gmaps-static"
+           ( "googlemaps"
+               :> "api"
+               :> "staticmap"
+               -- Why do we capture path segments here? We don't want to allow
+               -- access to the proxied API beyond the base path. (Who knows
+               -- what might be accessible then?!) The Handler will return HTTP
+               -- 404 if there are any illegal path segments.
+               :> Servant.CaptureAll "illegal_segments" String
```

Comment:
There is also Endpath in /libs/extended.  Just saw it by chance, not sure it's any nicer than your inlined solution?